### PR TITLE
Unborked auth

### PR DESCRIPTION
**Pull request by andricDu**

### CODE REVIEW

**Comment by andricDu** [comment]
Has a known bug with public key not being public. @Buwujiu is working on it.

File: src/main/java/bio/overture/ego/service/TokenService.java [hunk 1]
```diff
@@ -262,6 +262,19 @@ public User getTokenUserInfo(String token) {
...
+  }
+
+  public Application getTokenAppInfo(String token) {
+    try {
+      Claims body = getTokenClaims(token);
+      val tokenClaims =
+          TypeUtils.convertToAnotherType(body, AppTokenClaims.class, Views.JWTAccessToken.class);
+      return applicationService.getById(UUID.fromString(tokenClaims.getSub()));
+    } catch (JwtException | ClassCastException e) {
+      log.error("Issue handling application token: {}", token);
```
Comment (rtisma) [inline src/main/java/bio/overture/ego/service/TokenService.java]: token should be md5ed

File: src/main/java/bio/overture/ego/security/JWTAuthorizationFilter.java [hunk 2]
```diff
@@ -65,24 +69,58 @@ public void doFilterInternal(
...
-    val authentication =
-        new UsernamePasswordAuthenticationToken(
-            tokenService.getTokenUserInfo(removeTokenPrefix(tokenPayload)),
-            null,
-            new ArrayList<>());
+    UsernamePasswordAuthenticationToken authentication = null;
+    val body = tokenService.getTokenClaims(removeTokenPrefix(tokenPayload));
+    try {
+      // Test Conversion
+      TypeUtils.convertToAnotherType(body, UserTokenClaims.class, Views.JWTAccessToken.class);
```
Comment (rtisma) [inline src/main/java/bio/overture/ego/security/JWTAuthorizationFilter.java]: static import

**Comment by andricDu** [comment]
Thanks for the reviews @rtisma @d8660091

File: src/main/java/bio/overture/ego/service/TokenService.java [hunk 3]
```diff
@@ -257,11 +259,24 @@ public boolean isValidToken(String token) {
...
   public User getTokenUserInfo(String token) {
     try {
-      Claims body = getTokenClaims(token);
+      val body = getTokenClaims(token);
       val tokenClaims =
-          TypeUtils.convertToAnotherType(body, UserTokenClaims.class, Views.JWTAccessToken.class);
+          convertToAnotherType(body, UserTokenClaims.class, Views.JWTAccessToken.class);
       return userService.get(tokenClaims.getSub());
     } catch (JwtException | ClassCastException e) {
+      log.error("Issue handling user token: {}", new String(md5Digest(token.getBytes())));
```
Comment (rtisma) [inline src/main/java/bio/overture/ego/service/TokenService.java]: change to `Issue handling user token (MD5sum)`

File: src/main/java/bio/overture/ego/security/JWTAuthorizationFilter.java [hunk 4]
```diff
@@ -65,24 +72,57 @@ public void doFilterInternal(
...
-  private void authenticateUser(String tokenPayload) {
+  /**
+   * Responsible for authenticating a Bearer JWT into a user or application.
+   *
+   * @param tokenPayload The string representation of the Authorization Header with the token prefix
+   *     included
+   */
+  private void authenticateUserOrApplication(String tokenPayload) {
     if (!isValidToken(tokenPayload)) {
+      log.warn("Invalid token: {}", new String(md5Digest(tokenPayload.getBytes())));
```
Comment (rtisma) [inline src/main/java/bio/overture/ego/security/JWTAuthorizationFilter.java]: change to `Invalid token (MD5sum):`

**Pull request merged**